crates/ruff_linter/src/linter.rs

```diff
@@ -528,19 +519,21 @@ pub fn lint_fix<'a>(
         );
 
         if iterations == 0 {
-            parseable = result.error.is_none();
+            parseable = parsed.is_valid();
```

Comment:
Nit: the name parseable is a bit missleading because our parser can now parse all files. Maybe valid-syntax or just call parsed.errors().first() and if it is Some, then there's an error (merge it with the error emitting path below)